src/lib/edit_script.rs

```diff
@@ -45,7 +45,11 @@ use action::{ApplyAction, Delete, EditScript, Insert, Move, Update};
 use ast::{ArenaError, FromNodeId, NodeId, ToNodeId};
 use matchers::{EditScriptResult, MappingStore, MappingType};
 
-const TMP_ROOT: &str = "_____DIFFRACT_TMP_ROOT";
+/// Label to use when introducing temporary root nodes in an AST.
+///
+/// Temporary root nodes should be removed before the edit script generator
+/// completes. This constant is exposed for use in integration tests.
+pub const TMP_ROOT: &str = "_____DIFFRACT_TMP_ROOT";
```

Comment:
Any reason for quite so many underscores? ;)

Reply:
No clever reason, I wanted to be reasonably sure that we didn't clash with any other label that might have been used in the tree.

Reply:
We're fully in control of all the labels, right? Can any of them start with an underscore?

Reply:
Well, the labels come from the text of the original program, so they can usually start with an underscore (unless the lexer objects, I guess). In the grammars-with-whitespace  you introduce new node types `^~` and `~` to deal with these situations, but I didn't want to add anything special to the grammars (and not all of our grammars have whitespace nodes), so the "temporary root nodes" have a default type (i.e. the type is made by saying `Default::default()`) and this weird label.

Reply:
What I suggest is that you put a prefix on all nodes outside your control, so you can guarantee there can never be clashes. Or would that involve changing too many things?

Reply:
Well, that means iterating over the whole tree twice to add the prefixes and then remove them, which seems a bit much. I'm not sure whether there is a better solution?

Reply:
Well, how about whenever you create a node you at least check it doesn't clash with that special ID? Basically I have learnt from bitter experience that unprotected magic IDs eventually blow up in my face one day...